pkg/controller/storageversionmigrator/storageversionmigrator.go

```diff
@@ -264,13 +276,20 @@ func (svmc *SVMController) sync(ctx context.Context, key string) error {
 		return err
 	}
 
-	// ToDo: implement a mechanism to resume migration from the last migrated resource in case of a failure
-	// process storage migration
 	for _, obj := range resourceMonitor.Store.List() {
 		accessor, err := meta.Accessor(obj)
 		if err != nil {
 			return err
 		}
+		rvCmp, err := resourceversion.CompareResourceVersion(accessor.GetResourceVersion(), listResourceVersion)
+		if err != nil {
+			logger.V(4).Error(err, "Unable to compare the resource version of the resource", "namespace", accessor.GetNamespace(), "name", accessor.GetName(), "gvr", gvr.String(), "accessorRV", accessor.GetResourceVersion(), "listResourceVersion", listResourceVersion, "error", err.Error())
+			return err
```

Comment:
I don't think we can recover from this by retrying so should this also fail the migration?